### ISSUE TITLE
Next.js app with DataStore works locally but not when deployed to Amplify Hosting

### ISSUE THREAD

**Issue by tombray** [aws-amplify/amplify-js#10732]
### Before opening, please confirm:

- [X] I have checked to see if my question is addressed in the [FAQ](https://github.com/aws-amplify/amplify-hosting/blob/master/FAQ.md).
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-hosting/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-hosting/blob/master/CONTRIBUTING.md).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

### App Id

d8dd1y47jar7i

### AWS Region

us-west-2

### Amplify Hosting feature

Not Applicable

### Describe the bug

I've created a simple Next.js app that uses a DataStore.query to fetch some data I generated in Amplify Studio. 

It works locally but when deployed to Amplify Hosting it throws an exception before posting the query.

Here's the exception:

> TypeError: Cannot read properties of undefined (reading 'length')
>     at n.copyOf (zen-observable.js:444:2)
>     at uq (zen-observable.js:444:2)
>     at e.<anonymous> (zen-observable.js:444:2)
>     at zen-observable.js:444:2
>     at Object.next (zen-observable.js:444:2)
>     at s (zen-observable.js:444:2)

### Expected behavior

Here's how the page renders locally:

<img width="288" alt="image" src="https://user-images.githubusercontent.com/311060/205121264-98a71961-1e86-41e0-8a05-202301acf739.png">

### Reproduction steps

Here's my repo: https://github.com/tombray/amplify-test

The DataStore.query is in pages/index.js:

```
export default function Home() {
  const [stations, setStations] = useState([]);

  useEffect(() => {
    const getStations = async () => {
      console.log("before datastore query");
      try {
        const stations = await DataStore.query(Stations);
        console.log(stations);
        setStations(stations);
      } catch (e) {
        console.log(e);
      }
    };

    getStations();
  }, []);

```
Running that locally works for me but when I deploy it with Amplify Hosting, it throws an exception which you can see here: https://main.d8dd1y47jar7i.amplifyapp.com/

### Log output

<details>

```
# Put your logs below this line


```

</details>

**Comment by svidgen** [comment]
Hey @tombray,

Thanks for the sample app! In my testing with your sample, this looks like an issue with SWC being the default starting with `next@13`. You can try disabling SWC to see if that unblocks you.

```js
const nextConfig = {
  productionBrowserSourceMaps: true,
  reactStrictMode: true,

  // adding this fixes it for me:
  swcMinify: false,

};
```

We have an issue open to further investigate the SWC this here: #10595. I'll make note on #10595 that we have another repro here -- and specifically that this could be the new *default* experience on next 13.

Thanks for reporting it. Please let me know if the SWC flag helps.

**Comment by tombray** [comment]
Thanks, @svidgen. Setting swcMinify to false fixed the exception but that exposed another issue (perhaps with my code but I'm not sure). The first time I load the page I don't see any results and an empty array logged to the console, although I do see the graphql request is coming back with the correct results. If I refresh the page, the results are displayed. My hunch is that it's my bug (I'm a little rusty with Next) but if you have any thoughts I'd greatly appreciate it!

**Comment by tombray** [comment]
Aha! Thanks @svidgen !!

**Comment by dpilch** [comment]
A fix was released in [v5.0.7](https://github.com/aws-amplify/amplify-js/releases/tag/aws-amplify@5.0.7). `swcMinify` can be set to `true` after upgrading.
